packages/stores/src/account/base.ts

```diff
@@ -682,95 +693,185 @@ export class AccountStore<Injects extends Record<string, any>[] = []> {
     ) {
-      throw new Error("signAmino is not available in wallet");
+      throw new Error("signDirect is not available in wallet");
     }
 
     const accountFromSigner = (await wallet.offlineSigner.getAccounts()).find(
       (account) => account.address === signerAddress
     );
-
     if (!accountFromSigner) {
       throw new Error("Failed to retrieve account from signer");
     }
-
     const pubkey = encodePubkey(
       encodeSecp256k1Pubkey(accountFromSigner.pubkey)
     );
+    const txBodyEncodeObject = {
+      typeUrl: "/cosmos.tx.v1beta1.TxBody",
+      value: {
+        messages: messages,
+        memo: memo,
+      },
+    };
 
-    const signMode = SignMode.SIGN_MODE_LEGACY_AMINO_JSON;
-    const msgs = messages.map((msg) => {
-      const res: any = wallet?.signingStargateOptions?.aminoTypes?.toAmino(msg);
-      // Include the 'memo' field again because the 'registry' omits it
-      if (msg.value.memo) {
-        res.value.memo = msg.value.memo;
-      }
-      return res;
-    }) as AminoMsg[];
+    // TODO: fix this by estimating the fee
+    fee.amount = [{ denom: "uosmo", amount: "10000" }];
+    const txBodyBytes = wallet?.signingStargateOptions?.registry?.encode(
+      txBodyEncodeObject
+    ) as Uint8Array;
 
-    const signDoc = makeSignDocAmino(
-      msgs,
-      fee,
+    const gasLimit = Int53.fromString(String(fee.gas)).toNumber();
+    const authInfoBytes = makeAuthInfoBytes(
+      [{ pubkey, sequence }],
+      fee.amount,
+      gasLimit,
+      fee.granter,
+      fee.payer
+    );
+    const signDoc: any = makeSignDoc(
+      txBodyBytes,
+      authInfoBytes,
       chainId,
-      memo,
-      accountNumber,
-      sequence
+      accountNumber
     );
 
-    const { signature, signed } = await (wallet.client.signAmino
-      ? wallet.client.signAmino(
-          wallet.chainId,
-          signerAddress,
-          signDoc,
-          wallet.walletInfo?.signOptions
-        )
-      : (wallet.offlineSigner as unknown as OfflineAminoSigner).signAmino(
-          signerAddress,
-          signDoc
-        ));
-
-    const signedTxBody = {
-      messages: signed.msgs.map((msg) => {
-        const res: any =
-          wallet?.signingStargateOptions?.aminoTypes?.fromAmino(msg);
-        // Include the 'memo' field again because the 'registry' omits it
-        if (msg.value.memo) {
-          res.value.memo = msg.value.memo;
-        }
-        return res;
-      }),
-      memo: signed.memo,
-    };
-
-    const signedTxBodyEncodeObject = {
-      typeUrl: "/cosmos.tx.v1beta1.TxBody",
-      value: signedTxBody,
-    };
-
-    const signedTxBodyBytes = wallet?.signingStargateOptions?.registry?.encode(
-      signedTxBodyEncodeObject
+    const sessionKey = localStorage.getItem("SessionKey");
+    if (sessionKey == null) {
+      throw new Error("session key not found or broken");
+    }
+    const privateKey = new PrivKeySecp256k1(fromBase64(sessionKey));
```

Comment:
Will this be able to support `PrivKeySecp256r1`?

Reply:
It should if we edit the code to generate an R1 signature, not sure if it's needed atm